### ISSUE TITLE
Global state in Assets

### ISSUE THREAD

**Issue by gmethvin** [playframework/playframework#6354]
The `Assets` and `AssetsInfo` objects currently use the global application for configuration. It should be fairly simple to move that all into the `AssetsBuilder` class.

The `Assets` object can be kept around but we should move anything that uses global state into `AssetsBuilder`. We probably should deprecate the `Assets` object as well.

**Comment by wsargent** [comment]
This will have an effect on https://github.com/webjars/webjars-play, so pinging @jamesward

**Comment by jamesward** [comment]
The `webjars-play` stuff now uses DI so we should be good.

**Comment by akalinovskiy** [comment]
I suppose I've faced with related bug. If my controller uses reverse routing assets URL on initialization then `Assets.checkForMinified` is always true. I think the main cause of this problem is a global state in `Assets` because global stated is set only [after Guice initialization](https://github.com/playframework/playframework/blob/2.5.8/framework/src/play-server/src/main/scala/play/core/server/ProdServerStart.scala#L49).

https://github.com/playframework/playframework/blob/2.5.8/framework/src/play/src/main/scala/play/api/controllers/Assets.scala#L265
https://github.com/playframework/playframework/blob/2.5.8/framework/src/play/src/main/scala/play/api/controllers/Assets.scala#L84

```
class AssetsGlobalStateTest extends PlaySpec {
  val config = Configuration(
    "assets.server" -> "https://assets.server.com",
    "assets.checkForMinified" -> false)

  val injector = new GuiceInjectorBuilder()
    .in(Mode.Dev)
    .bindings(
      bind[Environment].toInstance(Environment.simple()),
      bind[Configuration].toInstance(config),
      bind[AssetsUser].toSelf)
    .injector()

  val assetsUser = injector.instanceOf[AssetsUser]

  "App" should {
    "AssetInfo.isDev == true" in { AssetInfo.isDev mustBe true }
    "checkForMinified == false" in { Assets.checkForMinified mustBe false }
  }
}

class AssetsUser {
  val url = routes.Assets.versioned("some-file")
}
```

**Comment by gmethvin** [comment]
Fixed by #6820
